python/python/lance/ray/distribute_task.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+from typing import Any, Callable, Dict, List, Optional
+
+import lance
+
+# ==============================================================================
+# Message Structure Constants
+# ==============================================================================
+TASK_ID_KEY = "task_id"
+PARTITION_KEY = "partition"
+
+# ==============================================================================
+# Data Component Keys
+# ==============================================================================
+FRAGMENT_KEY = "fragment"
+SCHEMA_KEY = "schema"
+
+# ==============================================================================
+# Operation Parameters
+# ==============================================================================
+PARAMS_KEY = "params"
+ACTION_KEY = "action"
+READ_COLUMNS_KEY = "read_columns"
+
+# ==============================================================================
+# Execution Metadata
+# ==============================================================================
+OPERATION_TYPE_KEY = "operation_type"
+VERSION_KEY = "version"
+
+
+class TaskInput:
+    """Container for task execution parameters and metadata."""
+
+    def __init__(
+        self,
+        task_id: str,
+        fn: Callable,
+        fragment: Any,
+        params: Optional[Dict[str, Any]] = None,
+    ):
+        self.task_id = task_id
+        self.fn = fn
+        self.fragment = fragment
+        self.params = params or {}
+
+
+class FragmentTask:
+    """Base class for distributed data processing tasks."""
+
+    def __init__(self, task_input: TaskInput):
+        self._fn = task_input.fn
+        self._task_id = task_input.task_id
+        self._fragment = task_input.fragment
+        self._params = task_input.params
+
+    def __call__(self) -> Dict[str, Any]:
+        """Execute task and return results in Arrow-serializable format."""
+        output = self._fn()
+        return {
+            TASK_ID_KEY: self._task_id,
+            PARTITION_KEY: {FRAGMENT_KEY: self._fragment, "output": output},
+        }
+
+
+class AddColumnTask(FragmentTask):
+    """Task for adding new columns to dataset fragments."""
+
+    def __init__(self, task_input: TaskInput):
+        super().__init__(task_input)
+        self._validate_input_params()
+        self.read_columns = self._params[READ_COLUMNS_KEY]
```

Comment:
Why read data from `_params` instead of

```
def __init__(self, task_input: TaskInput, read_columns: ???)
```

If we are going to keep this style can we do...

```
self.read_columns: TYPE = self._params[READ_COLUMNS_KEY]
```

This way it will be easier to know what the type of `read_columns` is.